### PR TITLE
Remove the concept of shard

### PR DESCRIPTION
**Pull request by sgkim126**
This PR reorders and re-structs the commits below.
- 815b01d1e Test module state from the top level
- 0077d001d Refactor module_level test utilizing test_helper
- f83cda57e Remove ShardTransaction
- 3726c34e1 Remove shard items
- 4a400cd0c Remove shard_cache
- 81e3981bb Remove shard_level state
- 33e2367ba Remove shard from Metadata
- f72062d50 Remove shard cache from TopCache and GlobalCache
- c3cb999b3 Modify SnapshotSync to include module db
- b5f58e403 Remove shard cache from the top level

### CODE REVIEW

**Comment by majecty** [comment]
@sgkim126  Could you tell me the decision about "removing shard"? I don't know it. I'll review this PR better if I know the decision.

**Comment by sgkim126** [comment]
@majecty Because the shard cannot work with the module, Foundry will not support shards anymore. If there is a module which wants to use shard, it's their responsibility to implement the shard internally.

**Comment by majecty** [comment]
Thanks. I'll review this PR now.

File: state/src/impls/test_helper.rs [hunk 1]
```diff
@@ -41,6 +41,61 @@ macro_rules! transaction {
...
+    ($state:expr, {set: $content:tt}) => {
+        set_module_level_state!($state, $content);
+    };
+    //recursion
+    ($state:expr, {$opl:ident: $contentl:tt, $($opr:ident: $contentr:tt),+ $(,)?}) => {
+        module_level!($state, {$opl: $contentl});
+        module_level!($state, {$($opr: $contentr),+});
+    }
+}
+
```
Comment (majecty) [inline state/src/impls/test_helper.rs:98]: These macros look good!

File: test/src/e2e/verification.test.ts [hunk 2]
```diff
@@ -144,8 +144,21 @@ describe("solo - 1 node", function() {
...
+            { actionType: 0x23, actionLength: 4 },
+            { actionType: 0x24, actionLength: 3 }, // SelfNominate
+            { actionType: 0x24, actionLength: 5 },
+            { actionType: 0x25, actionLength: 2 }, // ReportDoubleVote
+            { actionType: 0x25, actionLength: 4 },
+            { actionType: 0x26, actionLength: 4 }, // Redelegate
+            { actionType: 0x26, actionLength: 5 },
+            { actionType: 0xff, actionLength: 1 }, // ChangeParams
+            { actionType: 0xff, actionLength: 2 },
+            { actionType: 0xff, actionLength: 3 }
```
[the remark this answers is not in the document]
Comment (sgkim126) [reply test/src/e2e/verification.test.ts:161]: I extracted this and the below change to a new commit, "Enhance the verification tests".

**Comment by majecty** [comment]
Since the failed test is "discovery5", I'll merge this PR.

**Pull request merged**